rustls/src/crypto/hmac.rs

```diff
@@ -45,11 +46,38 @@ impl Tag {
 
 impl Drop for Tag {
     fn drop(&mut self) {
-        self.buf.zeroize();
+        self.0.buf.zeroize();
     }
 }
 
 impl AsRef<[u8]> for Tag {
+    fn as_ref(&self) -> &[u8] {
+        self.0.as_ref()
+    }
+}
+
+/// A HMAC tag, stored as a value, that is non-secret.
```

Comment:
Maybe add some context on what exactly it means for a tag to be public, and also why it matters/what's different about it?